src/lib.rs

```diff
@@ -39,6 +39,39 @@ pub fn parse_and_inline_modules(src_file: &Path) -> syn::File {
         .output
 }
 
+/// The string `syn_inline_mod_path`. This acts as a marker for inlined paths.
+pub static SYN_INLINE_MOD_PATH: &str = "syn_inline_mod_path";
+
+/// Given a list of attributes from an inlined module, search for the first attribute of the form
+/// `#[syn_inline_mod_path(b"...")]`. If it is found, then return a triple with:
+/// * the module path
+/// * the outer attributes
+/// * the inner attributes
+///
+/// See the documentation for `InlinerBuilder::annotate_paths` for more information.
+pub fn find_mod_path(attrs: &[Attribute]) -> Option<(PathBuf, &[Attribute], &[Attribute])> {
+    let mut path = None;
+    let position = attrs.iter().position(|attr| {
+        if attr.path.is_ident(SYN_INLINE_MOD_PATH) {
+            // Ignore the attribute if it isn't of the correct form.
```

Comment:
Is your worry here that some other proc-macro might happen to squat on our attribute name? Finding an attribute of the right name and wrong format feels to me like something must have gone horribly wrong.

Reply:
Yeah or the user puts the annotation in themselves or something. The options are to panic or ignore -- what do you think about them? I guess panicking is probably better/safer since it'd be totally unexpected to hit this.

Reply:
Well I realized this may end up dealing with external data so we'd probably rather not panic. I think this should be a pretty rare case though, as you pointed out something's really gone wrong (but we don't want to create an opportunity for an accidental DoS).